tox.ini

```diff
@@ -28,7 +31,7 @@ commands = flake8 pythonforandroid/ tests/ ci/
 
 [flake8]
 ignore =
-    E123, E124, E126,
+    E123, E124, E126, E127, E129
```

Comment:
I'm complying with the "a foolish consistency is the hobgoblin of little minds" part :<

I have:

    ```
with mock.patch('sys.version_info') as fake_version_info, \
     mock.patch('pythonforandroid.entrypoints.handle_build_exception') as handle_build_exception:

```

Flake8 would prefer misalignment:

```
with mock.patch('sys.version_info') as fake_version_info, \
         mock.patch('pythonforandroid.entrypoints.handle_build_exception') as handle_build_exception:
```

I think this is plainly less consistent and readable. In fact, it's exactly the form given in pep8 as an example of how with statements are an exception to the normal recommendations:

```
with open('/path/to/some/file/you/want/to/read') as file_1, \
     open('/path/to/some/file/being/written', 'w') as file_2:
    file_2.write(file_1.read())
```
How about if I make this line a #noqa?

Reply:
Interestingly, black makes a real mess of this:

```
    with mock.patch("sys.version_info") as fake_version_info, mock.patch(
        "pythonforandroid.entrypoints.handle_build_exception"
    ) as handle_build_exception:
```
I can't be the only one who finds this a much worse choice!

Reply:
Thanks for justifying it, yes it makes sense to me. I also hate the misalignment.
In fact I have two was to make it aligned while still complying with flake8. One involves blackslashes which @opacam [suggested he didn't like](https://github.com/kivy/python-for-android/pull/1928#discussion_r307172795), the other one involves parenthesis, which [you suggested you didn't like](https://github.com/kivy/python-for-android/pull/1946#discussion_r309430894) :smile: 
I agree with your statement of foolish consistency, however in general enforcing consistency systematically brings more good than not having it.
Yes I would rather use a noqa too in this case, but I'm not sure how it behaves on multiple lines.
Yes I also think black sucks in this area, another reason I hate it. But also it would avoid us having these discussion and focus on other things, that's the reason I love it.

Reply:
Well guys, I see more ways to this without relying on black... we can split lines in a lot of ways, this based on @AndreMiras solution in one of his recent PRs:
```
def mock_sys_info():
    return mock.patch('sys.version_info')


def mock_handle_exception():
    return mock.patch('pythonforandroid.entrypoints.handle_build_exception')


def test_main_python2():

    with mock_sys_info() as m_version, mock_handle_exception() as m_exception:

        m_version.major = 2
        m_version.minor = 7

        def check_python2_exception(exc):
            assert exc.message == PY2_ERROR_TEXT
        m_exception.side_effect = check_python2_exception

        entrypoints.main()

    m_exception.assert_called_once()
```

With this, you keep all the code below 80 characters, no `noqa` and using `Context manager`.

But, I would go with decorators...something like:

```
@mock.patch('pythonforandroid.entrypoints.handle_build_exception')
@mock.patch('sys.version_info')
def test_main_python2(fake_version_info, handle_build_exception):

    def check_python2_exception(exc):
        assert exc.message == PY2_ERROR_TEXT

    fake_version_info.major = 2
    fake_version_info.minor = 7
    handle_build_exception.side_effect = check_python2_exception

    entrypoints.main()

    handle_build_exception.assert_called_once()
```

Black it's great as an ortographic corrector and also to unify style of all the project even to find this kind of lines which doesn't fit as expected...then...IMHO... one must push the imagination a little further.

*Play what you know and then play above that*
**Miles Davis**